### ISSUE TITLE
Unit Tests?

### ISSUE THREAD

**Issue by joeaudette** [cloudscribe/cloudscribe#2]
You may notice there is very little at this point in the way of unit tests in this project. I acknowledge that I am not very experienced with TDD (Test Driven Development) and have not been working with that as my process. Much of the code was existing code migrated and refactored from mojoportal.

With this project I am trying to develop testable code and would like to add more and more test coverage over time. I would specifically welcome collaborators who are more experienced with TDD to help get the ball rolling by implementing tests, pointing out any areas where code is not testable currently, making pull requests that refactor code to become more testable, etc. I learn quickly from other people's examples and believe that with help and collaboration I could become proficient in TDD.

I'm not sure yet what mocking framework to use for example or which ones are compatible with the new .NET frameworks.

In short, please don't beat me up about lack of tests, instead help me move forward in making sure things are testable and adding to test coverage.

**Comment by trendoid** [comment]
I believe there is only one unit test framework fully compatible with ASP.NET 5 right now.  Xunit
I haven't looked at your tests but if you went down the same road I did a month or so ago you're probably transitioning a bit from Microsoft or Nunit to Xunit.  I will try to volunteer some input soon.  I have to look at my code from last month on mocking.  I will have time this weekend after hours.  I do think this is a really cool project you are undertaking.  I would love to help out.

**Comment by joeaudette** [comment]
I know about xunit you can see it in the few little tests already there. I think moq is probably what we will use for mocking but with things changing so fast in the core frameworks things break often. beta8 will bring a lot of breaking changes soon. even things that do support the frameworks can be broken by new framework releases at this point and then we have to wait for them to provide an update that is compatible with the new framework release. xunit has been relatively stable so far, but if I recall, last time I checked moq wasn't ready though it might be ready now.

**Comment by joeaudette** [comment]
wanted to update this issue just to mention that the few working unit tests we have for the pagination project have been moved along with that project into it's own repo [cloudscribe.Web.Pagination](https://github.com/joeaudette/cloudscribe.Web.Pagination)

So this repo no longer has any working unit tests though we have one project stubbed out for testing.

A good way to get involved with this project for anyone interested would be to write some tests and submit pull requests to add them to this repo. Contributing some meaningful, well written unit tests to get the ball rolling and provide examples that myself and others could follow would be a great way to gain favor with this project and move towards becoming a collaborator with commit access.
